src/main/java/bio/terra/service/filedata/azure/AzureSynapsePdao.java

```diff
@@ -0,0 +1,315 @@
+    String containerName = ingestControlFileBlobUrl.getBlobContainerName();
+
+    BlobContainerClientFactory targetDataClientFactory =
+        azureBlobStorePdao.getTargetDataClientFactory(
+            profileModel,
+            storageAccount,
+            containerName,
+            getPermissionString(SynapseSasPermission.WRITE_PARQUET));
+
+    // Given the sas token, rebuild a signed url
+    String signedURL =
+        targetDataClientFactory.createSasUrlForBlob(blobName, getPermissionString(permissionType));
+    return BlobUrlParts.parse(signedURL);
+  }
+
+  public void createExternalDataSource(
+      BlobUrlParts signedBlobUrl, String scopedCredentialName, String dataSourceName)
+      throws NotImplementedException, SQLException {
+    AzureSasCredential blobContainerSasTokenCreds =
+        new AzureSasCredential(signedBlobUrl.getCommonSasQueryParameters().encode());
+
+    ST sqlScopedCredentialCreateTemplate = new ST(scopedCredentialCreateTemplate);
+    sqlScopedCredentialCreateTemplate.add("scopedCredentialName", scopedCredentialName);
+    sqlScopedCredentialCreateTemplate.add("secret", blobContainerSasTokenCreds.getSignature());
+    executeSynapseQuery(sqlScopedCredentialCreateTemplate.render());
+
+    ST sqlDataSourceCreateTemplate = new ST(dataSourceCreateTemplate);
+    sqlDataSourceCreateTemplate.add("dataSourceName", dataSourceName);
+    sqlDataSourceCreateTemplate.add("scheme", signedBlobUrl.getScheme());
+    sqlDataSourceCreateTemplate.add("host", signedBlobUrl.getHost());
+    sqlDataSourceCreateTemplate.add("container", signedBlobUrl.getBlobContainerName());
+    sqlDataSourceCreateTemplate.add("credential", scopedCredentialName);
+    executeSynapseQuery(sqlDataSourceCreateTemplate.render());
+  }
+
+  public void createParquetFiles(
+      FormatEnum ingestType,
+      DatasetTable datasetTable,
+      String ingestFileName,
+      String destinationParquetFile,
+      String destinationDataSourceName,
+      String controlFileDataSourceName,
+      String ingestTableName,
+      int csvSkipLeadingRows)
+      throws SQLException {
+
+    List<Column> columns = datasetTable.getColumns();
+    String tableDefinition =
+        String.join(
+            ",\n      ",
+            columns.stream()
+                .map(
+                    c -> {
+                      if (ingestType == FormatEnum.CSV) {
+                        return c.getName() + " " + translateTypeToDdl(c.getType(), c.isArrayOf());
+                      } else if (ingestType == FormatEnum.JSON) {
+                        return translateToJsonConvert(c.getName(), c.getType(), c.isArrayOf());
+                      } else {
+                        throw new EnumConstantNotPresentException(
+                            FormatEnum.class, ingestType.toString());
+                      }
+                    })
+                .collect(Collectors.toList()));
+
+    ST sqlCreateTableCSVTemplate = new ST(createTableTemplate);
```

Comment:
It's fine for now.  Just a heads up that eventually, we're probably going to want a method that returns this sql for the access_info property in snapshots and datasets.  Like I said let's leave it like this for now but let's remember that we'll want to probably factor this out when we take on that story

Reply:
Ok, sounds good!